src/dataset/generator.rs

```diff
@@ -0,0 +1,153 @@
+//! # The Boston Housing Dataset
```

Comment:
header with documentation has been copied from _The Boston Housing Dataset_. Feel free to push it as is, I can update it. Functions `make_blobs ` and `make_circles ` are copied from Scikit-Learn, right?